Add new function to specify namespace for `KedroSession` and `kedro run` (#2306)
kedro-org/kedro: RELEASE.md

  * Save node outputs after every `yield` before proceeding with next chunk.
* Fixed incorrect parsing of Azure Data Lake Storage Gen2 URIs used in datasets.
* Added support for loading credentials from environment variables using `OmegaConfigLoader`.

> * Added new ability to filter by namespace for `KedroSession.run` and `kedro run` with `--namespace`.

Reviewer: ```suggestion
* Added new `--namespace` flag to `kedro run` to enable filtering by node namespace.
```
I don't think `KedroSession.run` is worth mentioning here just to keep release notes concise, since CLI `kedro run` is how 99% of kedro runs are implemented rather than the Python API.
